### DISCUSSION TITLE
Object storage usage increased by 100GB after using gc --compact

### DISCUSSION THREAD

**Discussion by FleetAdmiralButter** [juicedata/juicefs#2399] [Q&A] [answered]
Hi,

I've been using the JuiceFS CSI driver in my Kubernetes cluster for some time now.

However, I noticed that the object storage service (Backblaze B2) was reporting ~391GB in use.

I mounted the filesystem on a separate VM and ran `du -sh` in the root, and it reported only 356GB was actually in use by files.

So, I ran `juicefs gc <meta-url> --delete --compact` in order to try reducing the space used. This command took a few hours to complete, but it had the complete opposite effect and now I have nearly 500GB used in the bucket! Running `du -sh` again in the root of the JuiceFS mount folder confirms that only 356GB is actually in use by files.

Here's the output of the `juicefs gc --compact --delete` command:
```
Scanned chunks count: 13853 / 13853 [==============================================================]  done      
Compacted slices count: 11830                             
Compacted slices bytes: 97.53 GiB  (104723703920 Bytes)   
   Listed slices count: 78035                             
 Deleted pending count: 1                                 
 Scanned objects count: 177744 / 177744 [==============================================================]  done      
   Valid objects count: 173174                            
   Valid objects bytes: 444.99 GiB (477802180238 Bytes)   
  Leaked objects count: 28                                
  Leaked objects bytes: 60.09 MiB  (63009139 Bytes)       
 Skipped objects count: 4542                              
 Skipped objects bytes: 15.67 GiB  (16830775835 Bytes)    
2022/07/25 09:37:29.416524 juicefs[38430] <INFO>: scanned 177744 objects, 173174 valid, 28 leaked (63009139 bytes), 4542 skipped (16830775835 bytes) [gc.go:306]
```

I reran both `gc` and `fsck` as well just to make sure:

gc:
```
  Listed slices count: 77523                             
Scanned objects count: 177150 / 177150 [==============================================================]  done      
  Valid objects count: 171820                            
  Valid objects bytes: 456.75 GiB (490426399776 Bytes)   
 Leaked objects count: 0                                 
 Leaked objects bytes: 0.00 b     (0 Bytes)              
Skipped objects count: 5330                              
Skipped objects bytes: 3.13 GiB   (3356313952 Bytes)     
2022/07/25 11:30:56.238940 juicefs[38853] <INFO>: scanned 177150 objects, 171820 valid, 0 leaked (0 bytes), 5330 skipped (3356313952 bytes) [gc.go:306]
```

fsck:
```
  Found blocks count: 176577                            
  Found blocks bytes: 459.59 GiB (493482248776 Bytes)   
 Listed slices count: 76853                             
Scanned slices count: 76853 / 76853 [==============================================================]  done      
Scanned slices bytes: 459.59 GiB (493480217984 Bytes)   
   Lost blocks count: 0                                 
   Lost blocks bytes: 0.00 b     (0 Bytes) 
```

Output of `du -sh`:
```
~# du -sh /mnt/juicefs
356G    /mnt/juicefs
```

I've checked the .trash directory and it is pretty much empty (~4MB used).

Did I do something wrong running the `gc` command, and is there any way to reduce the "wasted" space?

My JuiceFS version is `juicefs version 1.0.0-rc2+2022-06-24.fc6b120`. This client version matches the installed CSI driver of v0.15.1.

**Comment by SandyXSD** [comment, marked as answer]
This is because compacted objects will be kept for `TrashDays` before auto cleaning up. You may disable trash by `juicefs config <META-URL> --trash-days 0` and then run `gc --delete` again to force remove them. Trash can be re-enabled with `config` command if you still need it.

**Reply by jiripospisil** [reply]
Could the amount of data in the trash be displayed somewhere?

**Reply by solracsf** [reply]
Yes, in the `.trash` folder. 
`du -sh .trash`

**Reply by SandyXSD** [reply]
For now, there is no direct way to see the amount of compacted objects. An issue is created to track this: https://github.com/juicedata/juicefs/issues/2411

**Reply by FleetAdmiralButter** [reply]
Thank you @SandyXSD! I've just checked the bucket again and usage has indeed gone down so it looks like the compacted objects have been cleaned up :)
